tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -933,6 +933,44 @@ func (az *AzureProvider) destroyDisk(ctx context.Context, vmInfo compute.Virtual
 	return
 }
 
+// DestroyWindowsVMWithResources destroys the Windows VMs along with all the errors associated with it.
+func (az *AzureProvider) DestroyWindowsVM(vmName string) error {
+	ctx := context.Background()
+	err, nicName := az.getNICname(ctx, vmName)
+	if err != nil {
+		log.Printf("error getting the nic name for %s: %v", vmName, err)
+	}
+	err, ipName := az.getIPname(ctx, vmName)
+	if err != nil {
+		log.Printf("error getting the ip name for the instance %s: %v", vmName, err)
+	}
+
+	err = az.destroyInstance(ctx, vmName)
+	if err != nil {
+		log.Printf("error destroying the instance %s: %v", vmName, err)
+	}
+
+	err = az.destroyNIC(ctx, nicName)
+	if err != nil {
+		log.Printf("error deleting the NIC of instance %s: %v", vmName, err)
+	}
+
+	err = az.destroyIP(ctx, ipName)
+	if err != nil {
+		log.Printf("error deleting the IP of instance %s: %v", vmName, err)
+	}
+	vmInfo, err := az.vmClient.Get(ctx, az.resourceGroupName, vmName, compute.InstanceView)
+	if err != nil {
+		log.Printf("error getting vminfo for %s: %v", vmName, err)
+	}
+	// This will still leak resources as we'll not be able to get vm info once the VM gets deleted.
```

Comment:
I dont understand this comment, what azure resources are being leaked here?

Reply:
Are you saying this will always fail?

Reply:
The disk deletion won't happen as it needs vminfo and vminfo would be nil for the deleted vm.

Reply:
Then shouldn't you move it before the VM is deleted? I don't understand why we are calling a function that is guaranteed to fail

Reply:
I just moved it back. Even if I move it, if the Delete operation is called on non-existent VM instead of failing hard where are proceeding further which will cause problems :(.

Reply:
It seems odd to me to call a function, expecting it to fail.